libs/framework/src/framework.c

```diff
@@ -640,7 +640,7 @@ celix_status_t fw_installBundle2(framework_pt framework, bundle_pt * bundle, lon
     }
 
     //increase use count of framework bundle to prevent a stop.
```

Comment:
The bundle count  changes so that bundles cannot be stopped/un-install when still in use (i.e. registering or a using a service) so that a bundle cannot be stopped/uninstalled when it is still in use. 

In this case the framework bundle use count is increased to ensure the framework cannot be stopped while it is installing another bundle.

The managing of the bundle use count is thread safe and the handling of bundle entries in the framework.c should also be thread safe, but the latter is difficult to say with 100% certainty (because it quite a lot and complex code).